source/common/upstream/eds_subscription_factory.h

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#include "envoy/api/v2/core/base.pb.h"
+#include "envoy/config/subscription.h"
+#include "envoy/stats/scope.h"
+
+#include "common/config/grpc_managed_mux_subscription_impl.h"
+#include "common/config/grpc_mux_impl.h"
+#include "common/config/subscription_factory.h"
+#include "common/config/utility.h"
+#include "common/protobuf/protobuf.h"
+
+namespace Envoy {
+namespace Upstream {
+class EdsSubscriptionFactory {
```

Comment:
A comment explaining what this does would be helpful, motivation, link back to original issue. Also a TODO explaining that it would make sense to generalize this for RDS in the future.